vspec/__init__.py

```diff
@@ -379,11 +379,21 @@ def expand_instances(flat_model):
     cont = True
 
     def extend_entry(new_entry, instance, name):
+        # It is expected that name exist in new_entry, either in "name" or "prefix", so two use-cases exist
+        # 1. Name part shall be refactored, e.g. Axle.TireAspectRatio -> Axle.Row1.TireAspectRatio
+        # 2. Prefix part shall be instantiated, e.g. Vehicle.Chassis.Axle.Wheel -> Vehicle.Chassis.Axle.Row2.Wheel
 
-        if base_name in new_entry["$name$"]:
+        if (name==new_entry["$name$"]) or new_entry["$name$"].startswith(name + "."):
             new_entry["$name$"] = new_entry["$name$"].replace(name, "{}.{}".format(name, instance), 1)
         else:
-            new_entry["$prefix$"] = new_entry["$prefix$"].replace(name, "{}.{}".format(name, instance), 1)
+            tmp = new_entry["$prefix$"]
+            search_str = "." + name
+            if tmp.endswith(search_str):
```

Comment:
This is some additional changes to try to avoid replacing the wrong part if the same term occurs earlier in the prefix.